### PR TITLE
Consolidate the bundles between manager and agent

### PR DESCRIPTION
**Pull request by yanmxa**
- Combine the manager and agent bundle into one entity
- Abstract the bundle interface to reduce the difficulty of the extensibility 
- Refactor the Conflation with the enhanced bundle to make it easier to read

### CODE REVIEW

File: pkg/bundle/cluster/hub_cluster_bundle.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+	bundle.lock.Lock()
+	defer bundle.lock.Unlock()
+
+	var routeURL string
+	route := object.(*routev1.Route)
+	if len(route.Spec.Host) != 0 {
+		routeURL = "https://" + route.Spec.Host
+	}
+
+	if len(bundle.Objects) == 0 {
```
Comment (clyang82) [inline pkg/bundle/cluster/hub_cluster_bundle.go:78]: You may get the latest code logic here. Your code changes are old.

**Review by clyang82 — COMMENTED** [review]
the basic ideas look good for me. combine the bundle together to reduce the learning curve effort.

it can be ready for merging once you fix the comment.

File: pkg/bundle/cluster/hub_cluster_bundle.go (hunk 1 above)
Comment (yanmxa) [reply pkg/bundle/cluster/hub_cluster_bundle.go]: Done.

**Pull request merged**